src/S3/Parser/S3Parser.php

```diff
@@ -0,0 +1,256 @@
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Checks if the root element of the response body is "Error", which is
+     * when we should try to parse an error from a 200 response from s3.
+     * It is recommended to make sure the stream given is seekable, otherwise
+     * the rewind call will cause a user warning.
+     * @param StreamInterface $responseBody
+     *
+     * @return bool
+     */
+    private function isFirstRootElementError(StreamInterface $responseBody): bool
+    {
+        static $pattern = '/<\?xml version="1\.0" encoding="UTF-8"\?>\s*<Error>/';
+        // To avoid performance overhead in large streams
+        $reducedBodyContent = $responseBody->read(64);
+        $foundErrorElement = preg_match($pattern, $reducedBodyContent);
+        // A rewind is needed because the stream is partially or entirely consumed
+        // in the previous read operation.
+        $responseBody->rewind();
+
+        return $foundErrorElement;
+    }
+
+    /**
+     * Execute mutator implementations over a result.
+     * Mutators are logics that modifies a result.
+     *
+     * @param ResultInterface $result
+     * @param CommandInterface $command
+     * @param ResponseInterface $response
+     *
+     * @return ResultInterface
+     */
+    private function executeS3ResultMutators(
+        ResultInterface $result,
+        CommandInterface $command,
+        ResponseInterface $response
+    ): ResultInterface
+    {
+        foreach ($this->s3ResultMutators as $mutator) {
+            $result = $mutator($result, $command, $response);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Adds a mutator into the list of mutators.
+     *
+     * @param $mutatorName
+     * @param S3ResultMutator $s3ResultMutator
+     * @return void
+     */
+    public function addS3ResultMutator($mutatorName, S3ResultMutator $s3ResultMutator): void
+    {
+        if (isset($this->s3ResultMutators[$mutatorName])) {
+            trigger_error("The S3 Result Mutator {$mutatorName} already exists!", E_USER_WARNING);
+
+            return;
+        }
+
+        $this->s3ResultMutators[$mutatorName] = $s3ResultMutator;
+    }
+
+    /**
+     * Removes a mutator from the mutator list.
+     *
+     * @param $mutatorName
+     * @return void
+     */
+    public function removeS3ResultMutator($mutatorName): void
+    {
+        if (!isset($this->s3ResultMutators[$mutatorName])) {
+            trigger_error("The S3 Result Mutator {$mutatorName} does not exist!", E_USER_WARNING);
+
+            return;
+        }
+
+        unset($this->s3ResultMutators[$mutatorName]);
+    }
+
+    /**
+     * Returns the list of result mutators available.
+     *
+     * @return array
+     */
+    public function getS3ResultMutators(): array
+    {
+        return $this->s3ResultMutators;
+    }
+
+    public function parseMemberFromStream(StreamInterface $stream, StructureShape $member, $response)
```

Comment:
line length

Reply:
it appears this wasn't addressed.  Break to a new line— this is over 80 characters